source/darwin/commoncrypto_aes.c

```diff
@@ -0,0 +1,701 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/cal/symmetric_cipher.h>
+
+#include <CommonCrypto/CommonCryptor.h>
+#include <CommonCrypto/CommonHMAC.h>
+#include <CommonCrypto/CommonSymmetricKeywrap.h>
+
+#include <aws/cal/private/darwin/common_cryptor_spi.h>
+
+#if defined(__MAC_OS_X_VERSION_MAX_ALLOWED)
```

Comment:
shouldn't we be checking based on ~MAX~ -> `MIN_ALLOWED` ?

I know we compile the CRT sometimes with OSX_DEPLOYMENT_TARGET=10.9

the [cmake docs](https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_DEPLOYMENT_TARGET.html#variable:CMAKE_OSX_DEPLOYMENT_TARGET) make it seems like this ends up setting a min (no mention of a max)

>  CMake uses this variable value for the -mmacosx-version-min flag

Reply:
changed the names, but this matches the exact match already done in the C++ SDK. I'll come back to it before merging.